lib/src/block_syntaxes/table_syntax.dart

```diff
@@ -71,29 +72,39 @@ class TableSyntax extends BlockSyntax {
   }
 
   List<String?> _parseAlignments(String line) {
-    final startIndex = _walkPastOpeningPipe(line);
+    final columns = <String?>[];
+    var started = false;
+    var hitDash = false;
+    String? align;
+
+    for (var i = 0; i < line.length; i++) {
+      final char = line.codeUnitAt(i);
+      if (char == $space || char == $tab || (!started && char == $pipe)) {
+        continue;
+      }
+      started = true;
 
-    var endIndex = line.length - 1;
-    while (endIndex > 0) {
-      final ch = line.codeUnitAt(endIndex);
-      if (ch == $pipe) {
-        endIndex--;
-        break;
+      if (char == $colon) {
+        if (hitDash) {
+          align = align == 'left' ? 'center' : 'right';
+        } else {
+          align = 'left';
+        }
+      } else {
+        hitDash = true;
```

Comment:
Hmm, we say `hitDash = true` if `char` is not `$colon`? I don't understand, should we rename this variable?

Reply:
Sorry, I put the hitDash at a wrong place. Fixed